File: frontend/lib/model/project/command/CreateProjectCommand.ts
export default class CreateProjectCommand {

    private readonly name: string;
    private readonly visibility: string;
    private readonly language: string;
    private readonly description: string;

    constructor(name: string, visibility: string, language: string, description: string) {
        this.name = name;

        this.visibility = visibility;

        this.language = language;
        this.description = description;
    }

}

File: backend/src/main/java/de/garrafao/phitag/application/project/data/CreateProjectCommand.java
package de.garrafao.phitag.application.project.data;

import org.apache.commons.lang3.Validate;

import lombok.Getter;

@Getter
public class CreateProjectCommand {
 
    private final String name;

    private final String visibility;
    
    private final String language;
    private final String description;

    public CreateProjectCommand(final String name, final String visibility, final String language, final String description) {
        Validate.notBlank(name, "name must not be blank");
        Validate.notBlank(visibility, "name must not be blank");
        Validate.notBlank(language, "language must not be blank");

        this.name = name;
        
        this.visibility = visibility;

        this.language = language;
        this.description = description;
    }

}
